frontend/src/app/testResults/DownloadResultsCsvButton.tsx

```diff
@@ -0,0 +1,182 @@
+import { useRef, useState } from "react";
+import Modal from "react-modal";
+import { faDownload } from "@fortawesome/free-solid-svg-icons";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { CSVLink } from "react-csv";
+import moment from "moment";
+
+import { displayFullName, showError } from "../utils";
+import { useImperativeQuery } from "../utils/hooks";
+import Button from "../commonComponents/Button/Button";
+import { TEST_RESULT_DESCRIPTIONS } from "../constants";
+import { symptomsStringToArray } from "../utils/symptoms";
+
+import {
+  byDateTested,
+  FilterParams,
+  hasSymptoms,
+  Results,
+  ResultsQueryVariables,
+  testResultQuery,
+} from "./TestResultsList";
+
+interface Props {
+  filterParams: FilterParams;
+  totalEntries: number;
+  facilityId: string;
+}
+
+const DownloadResultsCSVButton = ({
+  filterParams,
+  totalEntries,
+  facilityId,
+}: Props) => {
+  const [modalIsOpen, setModalIsOpen] = useState(false);
+  const [results, setResults] = useState([]);
+  const [loading, setLoading] = useState(false);
+  const csvLink = useRef<
+    CSVLink & HTMLAnchorElement & { link: HTMLAnchorElement }
+  >(null);
+
+  const openModal = () => setModalIsOpen(true);
+  const closeModal = () => setModalIsOpen(false);
+
+  const filtersPresent =
+    filterParams.patientId ||
+    filterParams.startDate ||
+    filterParams.endDate ||
+    filterParams.role ||
+    filterParams.result;
+
+  const variables: ResultsQueryVariables = {
+    facilityId,
+    pageNumber: 0,
+    pageSize: totalEntries,
+    ...filterParams,
+  };
+
+  const getResults = useImperativeQuery(testResultQuery, {
+    fetchPolicy: "no-cache",
+  });
+
+  const parseDataForCSV = (data: any) =>
+    data.sort(byDateTested).map((r: any) => {
+      const symptomList = r.symptoms ? symptomsStringToArray(r.symptoms) : [];
+      return {
+        "Patient first name": r.patient.firstName,
+        "Patient middle name": r.patient.middleName,
+        "Patient last name": r.patient.lastName,
+        "Patient full name": displayFullName(
+          r.patient.firstName,
+          r.patient.middleName,
+          r.patient.lastName
+        ),
+        "Patient date of birth": moment(r.patient.birthDate).format(
```

Comment:
This is probably more a question for @LGFranklin and @Rebecca-LM, but I'm wondering if we should adjust the headers to include the data format of the dates - so "Patient date of birth (MM/DD/YYYY)" instead of just "Patient date of birth"